ibis/backends/pyspark/compiler.py

```diff
@@ -140,15 +141,25 @@ def compile_selection(t, expr, scope, timecontext, **kwargs):
             ]
             col_in_selection_order.extend(cols)
         elif isinstance(selection, (types.ColumnExpr, types.ScalarExpr)):
-            col = t.translate(selection, scope, combined_timecontext).alias(
-                selection.get_name()
-            )
-            col_in_selection_order.append(col)
+            # If the selection is a straightforward projection of a table
+            # column from the root table itself (i.e. excluding mutations and
+            # renames), we can get the selection name directly.
+            if (
+                isinstance(selection.op(), ops.TableColumn)
+                and selection.op().table == op.table
+                and selection.get_name() in op.table.schema()
```

Comment:
Why do we need these two checks?

```
and selection.get_name() in op.table.schema()
and selection.op() == op.table[selection.get_name()].op()
```

Reply:
Can you perhaps also explain in the comment why do we need this?

Reply:
Checking that a) the selection name exists in the root table and b) its op exists in the root table, are both needed to fully qualify the selection as a 'pure' projection so that we only enter this new branch when appropriate -- i.e. it rules out a mutation that 1) assigns a new column or 2) changes an existing column. For example, if we did:
`table = table.mutate(new_col=table['A'])
table = table.mutate(new_col=table['B'])
`
we would need that additional check of `and selection.op() == op.table[selection.get_name()].op()` to avoid doing a simple projection of 'new_col' when what we really need to do is translate this selection since it's assigning a new value to an existing column.

Reply:
I see. This is fine then.

Perhaps we can abstract this logic to a function:

```
def can_be_replaced_by_column_name(column_expr, table):
# test if the column_expr can be replaced by it's name, i.e.:
# column_expr and table[column_expr.get_name()] is the semantically the same

```